File: core/src/main/java/net/dontdrinkandroot/wicket/model/CssClassClassStringModel.java
package net.dontdrinkandroot.wicket.model;

import net.dontdrinkandroot.wicket.css.CssClass;

import org.apache.wicket.model.IModel;
import org.apache.wicket.model.Model;


public class CssClassClassStringModel extends AbstractChainedModel<CssClass, String> {

	public CssClassClassStringModel(IModel<? extends CssClass> parent) {

		super(parent);
	}


	public CssClassClassStringModel(CssClass cssClass) {

		super(new Model<CssClass>(cssClass));
	}


	@Override
	public String getObject() {

		if (this.getParentObject() == null || !this.isActive()) {
			return null;
		}

		return this.getParentObject().getClassString();
	}


	protected boolean isActive() {

		return true;
	}

}

File: core/src/main/kotlin/net/dontdrinkandroot/wicket/model/CssClassClassStringModel.kt
package net.dontdrinkandroot.wicket.model

import net.dontdrinkandroot.wicket.css.CssClass
import org.apache.wicket.model.IModel

class CssClassClassStringModel : AbstractChainedModel<CssClass?, String?> {

    var active: Boolean = true

    constructor(parent: IModel<out CssClass?>) : super(parent)

    constructor(cssClass: CssClass) : super(cssClass.model())

    override fun getValue(parentValue: CssClass?): String? =
        if (parentValue == null || !active) null else parentValue.classString

}